Sources/IO/Geometry/GLTFReader/Animation/AnimationTrack.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable operator-assignment */
+/* eslint-disable no-param-reassign */
+import { quat, vec3 } from 'gl-matrix';
+
+class AnimationTrack {
+  constructor(target, path, times, values, interpolation) {
+    this.target = target;
+    this.path = path;
+    this.times = times;
+    this.values = values;
+    this.interpolation = interpolation;
+    this.duration = times[times.length - 1];
+    this.componentsPerValue = this.getComponentsPerValue();
+  }
+
+  getComponentsPerValue() {
+    switch (this.path) {
+      case 'translation':
+      case 'scale':
+        return 3;
+      case 'rotation':
+        return 4;
+      default:
+        console.warn(`Unsupported animation path: ${this.path}`);
+        return 0;
+    }
+  }
+
+  getValue(time) {
+    time = time % this.duration; // Loop the animation
+
+    let i = 1;
+    for (; i < this.times.length; i++) {
+      if (this.times[i] > time) break;
+    }
+    i--;
+
+    const t = (time - this.times[i]) / (this.times[i + 1] - this.times[i]);
+
+    const startIndex = i * this.componentsPerValue;
+    const endIndex = (i + 1) * this.componentsPerValue;
+
+    const value1 = this.values.subarray(
+      startIndex,
+      startIndex + this.componentsPerValue
+    );
+    const value2 = this.values.subarray(
+      endIndex,
+      endIndex + this.componentsPerValue
+    );
+
+    switch (this.interpolation) {
+      case 'STEP':
+        return value1;
+      case 'CUBICSPLINE':
+        console.warn('Cubic spline interpolation not implemented');
+        return this.linearInterpolate(value1, value2, t);
+      case 'LINEAR':
+      default:
+        return this.linearInterpolate(value1, value2, t);
+    }
+  }
+
+  linearInterpolate(a, b, t) {
+    switch (this.path) {
```

Comment:
instead of evaluating `this.path` each time, you could do in AnimationTRack constructor:
```
switch (this.path) { 
 ... 
 this.interpFunc = (a, b, t) => vec3.lerp(vec3.create(), a, b, t);`
```

and in `getValue()`:
```
return this.interpFunc(value1, value2, t)
```